src/views/token-swap/components/swap-button/swap-button.view.jsx

```diff
@@ -4,20 +4,19 @@ import PropTypes from 'prop-types'
 import FormButton from '../../../shared/form-button/form-button.view'
 import useSwapButtonStyles from './swap-button.style'
 
-function SwapButton ({ quotes }) {
+function SwapButton ({
+  quotes,
+  selectedQuote
+}) {
   const classes = useSwapButtonStyles()
-  const [timeUntilValid, setTimeUntilValid] = React.useState(30000)
+  const [timeUntilValid, setTimeUntilValid] = React.useState(selectedQuote?.validUntil.getTime())
```

Comment:
Why do we need this state variable? I think that we are only using it inside the effect to set the `countdown` right?